### PR TITLE
Fix Commits nil pointer dereference

### PR DESCRIPTION
**Pull request by Bwko**

### CODE REVIEW

**Comment by andreynering** [comment]
LGTM

**Comment by ethantkoenig** [comment]
LGTM

**Comment by lunny** [comment]
There is no commit? We cannot simply return an 404 and as an error but a page.

**Comment by Bwko** [comment]
@lunny What do you mean? I don't get it

File: routers/repo/commit.go [hunk 1]
```diff
@@ -47,6 +47,10 @@ func renderIssueLinks(oldCommits *list.List, repoLink string) *list.List {
 // Commits render branch's commits
 func Commits(ctx *context.Context) {
 	ctx.Data["PageIsCommits"] = true
+	if ctx.Repo.Commit == nil {
+		ctx.Handle(404, "Commit not found", nil)
```
Comment (lunny) [inline routers/repo/commit.go:51]: If `ctx.Repo.Commit == nil`, is that an error? Why return 404?

Comment (Bwko) [reply routers/repo/commit.go:51]: I think so, somehow my repo got corrupt and some commits are gone. 
Resulting in `ctx.Repo.Commit == nil`

**Comment by lafriks** [comment]
@lunny are you ok with this?

**Pull request merged**